### ISSUE TITLE
SC2317 findings are confusing and difficult to act on

### ISSUE THREAD

**Issue by dimo414** [koalaman/shellcheck#2613]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2317
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

foo() {
  bar() { baz; }
}
```

#### Here's what shellcheck currently says:

```
$ shellcheck myscript

Line 4:
  bar() { baz; }
          ^-- [SC2317](https://www.shellcheck.net/wiki/SC2317) (info): Command appears to be unreachable. Check usage (or ignore if invoked indirectly).
```

#### Here's what I wanted or expected to see:

First off, this control flow analysis seems super cool! That said, the findings it emits at present are pretty confusing to work with.

In my specific case, passing [bash-cache.sh](https://github.com/dimo414/bash-cache) to ShellCheck at present triggers ~50 SC2317 findings, none of which appear to be accurate/actionable. For example, the first two findings are on [this line](https://github.com/dimo414/bash-cache/blob/c9fa89fb9141660167c1cafa59ece39654b367a9/bash-cache.sh#L281), on the `(` and `{`. 

Now bash-cache does some unusual things like defining dynamic functions, so it's not totally surprising that some of it confuses control-flow analysis, but even so these findings don't seem very actionable to me. In particular, it's not at all clear from the finding how the reachability was determined or why the marked lines are "unreachable", which makes it hard to determine if there's anything to do about it other than suppress.

Furthermore, the fact that the error message itself includes guidance to ignore the finding and the wiki directs users to suppress it on entire functions or files suggests this is generating a lot of false positives. IMHO a check like this should err on the side of false negatives rather than introducing noise with false positives that users then have to suppress.
